### PR TITLE
Address internal error in localhost when adding ChatBot

### PR DESCRIPTION
**Pull request by ewpatton**
Change-Id: I3137482dd11767872b0cd3dea06e4c2e180a7a82

### CODE REVIEW

**Comment by ewpatton** [comment]
Note that I'm submitting this as a draft for feedback. If we decide to move forward with this change we should apply the same logic to the CloudDB, ImageBot, and Translator tokens.

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChatBot.java [hunk 1]
```diff
@@ -107,13 +107,19 @@ private void getTokenFromServer() {
...
       public void onSuccess(String token) {
-        EditableProperty tokenProperty = MockChatBot.this.properties.getProperty(PROPERTY_NAME_TOKEN);
-        if (tokenProperty != null) {
-          String existingToken = tokenProperty.getValue();
-          if (!existingToken.isEmpty()) {
-            LOG.info("bailing on getTokenFromServer existingToken = " + existingToken);
-            return;             // If we have a value, don't over-write it
-          }
+        if (token == null) {
+          onFailure(new UnsupportedOperationException(
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChatBot.java:111]: One thing to think about is whether we want to internationalize this message or not. In theory, this error indicates that the server is misconfigured and shouldn't happen in production, but famous last words and such.

Comment (SusanRatiLane) [reply appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChatBot.java:111]: > One thing to think about is whether we want to internationalize this message or not. In theory, this error indicates that the server is misconfigured and shouldn't happen in production, but famous last words and such.

I would say not to internationalize it. It doesn't seem like a good use of translator effort for something that is not expected to be user-facing.

**Review by SusanRatiLane — CHANGES_REQUESTED** [review]
In my tests on localhost, this is throwing an null pointer exception when attempting to change a property value. The code here only seems to be attempting to retrieve a property value, so I'm not entirely clear on the issue. I'm not sure that all expected errors are being caught.

**Comment by ewpatton** [comment]
@SusanRatiLane I am not seeing the NPE error you report. Maybe we can look at it when we are both in the office to track down what is happening.

**Review by SusanRatiLane — APPROVED** [review]
Revisited the error and determined it was a configuration error on my end.

**Comment by ewpatton** [comment]
@SusanRatiLane I've updated the PR to address the other components with token auth. I've also taken this out of draft mode for a formal review.

**Review by SusanRatiLane — APPROVED** [review]
LGTM

**Pull request merged**